proxy_manager.go

```diff
@@ -45,61 +45,115 @@ func newProxyManager(bundle creationBundle) *proxyManager {
 }
 
 func (m *proxyManager) getMap(ctx context.Context, name string) (*Map, error) {
-	if p, err := m.proxyFor(ctx, ServiceNameMap, name); err != nil {
+	if p, err := m.proxyFor(ctx, ServiceNameMap, name, true); err != nil {
 		return nil, err
 	} else {
 		return newMap(p), nil
 	}
 }
 
 func (m *proxyManager) getReplicatedMap(ctx context.Context, name string) (*ReplicatedMap, error) {
-	if p, err := m.proxyFor(ctx, ServiceNameReplicatedMap, name); err != nil {
+	if p, err := m.proxyFor(ctx, ServiceNameReplicatedMap, name, true); err != nil {
 		return nil, err
 	} else {
 		return newReplicatedMap(p, m.refIDGenerator)
 	}
 }
 
 func (m *proxyManager) getQueue(ctx context.Context, name string) (*Queue, error) {
-	if p, err := m.proxyFor(ctx, ServiceNameQueue, name); err != nil {
+	if p, err := m.proxyFor(ctx, ServiceNameQueue, name, true); err != nil {
 		return nil, err
 	} else {
 		return newQueue(p)
 	}
 }
 
 func (m *proxyManager) getTopic(ctx context.Context, name string) (*Topic, error) {
-	if p, err := m.proxyFor(ctx, ServiceNameTopic, name); err != nil {
+	if p, err := m.proxyFor(ctx, ServiceNameTopic, name, true); err != nil {
 		return nil, err
 	} else {
 		return newTopic(p)
 	}
 }
 
 func (m *proxyManager) getList(ctx context.Context, name string) (*List, error) {
-	if p, err := m.proxyFor(ctx, ServiceNameList, name); err != nil {
+	if p, err := m.proxyFor(ctx, ServiceNameList, name, true); err != nil {
 		return nil, err
 	} else {
 		return newList(p)
 	}
 }
 
 func (m *proxyManager) getSet(ctx context.Context, name string) (*Set, error) {
-	if p, err := m.proxyFor(ctx, ServiceNameSet, name); err != nil {
+	if p, err := m.proxyFor(ctx, ServiceNameSet, name, true); err != nil {
 		return nil, err
 	} else {
 		return newSet(p)
 	}
 }
 
 func (m *proxyManager) getPNCounter(ctx context.Context, name string) (*PNCounter, error) {
-	if p, err := m.proxyFor(ctx, ServiceNamePNCounter, name); err != nil {
+	if p, err := m.proxyFor(ctx, ServiceNamePNCounter, name, true); err != nil {
 		return nil, err
 	} else {
 		return newPNCounter(p), nil
 	}
 }
 
+func (m *proxyManager) invokeOnRandomTarget(ctx context.Context, request *proto.ClientMessage, handler proto.ClientMessageHandler) (*proto.ClientMessage, error) {
+	p, err := m.createProxy(ctx, "", "", false)
```

Comment:
I couldn't decide if it's a good idea to create an unnamed proxy as a field of the manager just for this purpose - not sure if any other api possibly needs it. Suggestions on this manner are welcome.